### PR TITLE
Makes red alert security level live up to its name

### PR DESCRIPTION
**Pull request by RipGrayson**
## About The Pull Request

Have you ever wondered why during red alert there's no theming at all? If you didn't see the announcement when you might not even know that the ship is under a dangerous threat at all.

This PR fixes that, when the ship changes to red security level, all lights on the ship gain a red tint. (as well as a minor reduction in intensity for atmosphere). In addition, changing to delta will also sound self destruct alarms through the ship until either the alert is lowered or the xenos manage to knock out power.

~~I also fixed like five lines of code that were keeping icons from displaying correctly for tables~~

## Why It's Good For The Game

The effect is cool and really gives the impression that the ship is under attack. It is a shame that currently the ship can be under attack by xeno forces and there's no blaring alarms or overt indication that things have gone wrong.

## Changelog
:cl:
expansion: Red alert now tints all ship-side lights red while active.
expansion: During Delta alert levels, self destruct alarms will sound.
fix: Fixed fancy tables not displaying correctly
fix: Fixed invisible blockers being visible
fix: Fixed a busted overlay check when welding scrubbers.
soundadd: Added a warning sound for self destruct during Delta alert levels.
imageadd: Added some custom icons for red tinted lights during red alert.
/:cl:

### CODE REVIEW

**Review by Kuro020 — APPROVED** [review]

**Comment by goose4429** [comment]
No hell march music 0/10
https://www.youtube.com/watch?v=e3YzmjmAGoI

**Comment by Derrick2Electricboogaloo** [comment]
Very nice. I have always wanted something such as this. I want to feel as if shit is about to go really bad

**Comment by FosterGrime-Heymynameslouie** [comment]
based pr spede merhg

File: code/datums/marine_main_ship.dm [hunk 1]
```diff
@@ -31,6 +31,46 @@ GLOBAL_DATUM_INIT(marine_main_ship, /datum/marine_main_ship, new)
...
 			level = SEC_LEVEL_DELTA
 
+	if(level <= SEC_LEVEL_BLUE)
+		for(var/obj/machinery/light/mainship/light AS in GLOB.mainship_lights)
+			light.base_state = "tube"
+			light.lightalarm.stop(light)
+			var/area/A = get_area(light)
+			if(!A.power_light || light.status != LIGHT_OK) //do not adjust unpowered or broken bulbs
+				continue
+			light.light_color = "#ffffff"
```
Comment (TiviPlus) [inline code/datums/marine_main_ship.dm:41]: use color defines please

File: code/datums/marine_main_ship.dm [hunk 2]
```diff
@@ -41,7 +81,7 @@ GLOBAL_DATUM_INIT(marine_main_ship, /datum/marine_main_ship, new)
 				security_level = SEC_LEVEL_GREEN
 				for(var/obj/machinery/status_display/SD in GLOB.machines)
 					if(is_mainship_level(SD.z))
-						SD.set_picture("default")
+						SD.set_picture("default")
```
Comment (TiviPlus) [inline code/datums/marine_main_ship.dm:84]:
```suggestion
						SD.set_picture("default")
```

File: code/datums/marine_main_ship.dm [hunk 3]
```diff
@@ -72,7 +112,7 @@ GLOBAL_DATUM_INIT(marine_main_ship, /datum/marine_main_ship, new)
 
 				for(var/obj/machinery/status_display/SD in GLOB.machines)
 					if(is_mainship_level(SD.z))
-						SD.set_picture("redalert")
+						SD.set_picture("redalert")
```
Comment (TiviPlus) [inline code/datums/marine_main_ship.dm:115]:
```suggestion
						SD.set_picture("redalert")
```

File: code/datums/marine_main_ship.dm [hunk 4]
```diff
@@ -86,7 +126,7 @@ GLOBAL_DATUM_INIT(marine_main_ship, /datum/marine_main_ship, new)
 
 		for(var/obj/machinery/firealarm/FA in GLOB.machines)
 			if(is_mainship_level(FA.z))
-				FA.update_icon()
+				FA.update_icon()
```
Comment (TiviPlus) [inline code/datums/marine_main_ship.dm:129]:
```suggestion
				FA.update_icon()
```

File: code/game/objects/effects/misc.dm [hunk 5]
```diff
@@ -72,7 +72,12 @@
 	opacity = FALSE
 	density = TRUE
 	icon_state = "blocker"
+	mouse_opacity = 0
```
Comment (TiviPlus) [inline code/game/objects/effects/misc.dm:75]: use the opacity define please

File: code/modules/power/lighting.dm [hunk 6]
```diff
@@ -146,6 +146,37 @@
 	var/switchcount = 0
 	/// true if rigged to explode
 	var/rigged = FALSE
+	///looping sound for red alert alarm sound
+	var/datum/looping_sound/alarm_loop/lightalarm
+
+/obj/machinery/light/mainship/
```
Comment (TiviPlus) [inline code/modules/power/lighting.dm:152]:
```suggestion
/obj/machinery/light/mainship
```

File: code/modules/power/lighting.dm [hunk 7]
```diff
@@ -146,6 +146,37 @@
 	var/switchcount = 0
 	/// true if rigged to explode
 	var/rigged = FALSE
+	///looping sound for red alert alarm sound
+	var/datum/looping_sound/alarm_loop/lightalarm
```
Comment (TiviPlus) [inline code/modules/power/lighting.dm:150]: This isnt very performant and is probably going to overlap and maybe be earrapey

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Pull request merged**